great_expectations/data_context/store/datasource_store.py

```diff
@@ -85,7 +86,20 @@ def serialize(self, value: FluentDatasource) -> dict:
         """
         See parent 'Store.serialize()' for more information
         """
-        return value._json_dict()
+        # DataAsset.order_by is not supported by v1, but has not yet been removed,
+        # so we drop it before serialization and alert the user with a warning.
```

Comment:
Why do we need this warning period before removing it?

Reply:
this will not be removed prior to release, but we're not serializing it, so this alerts the user